parsers/CA_BC.py

```diff
@@ -1,17 +1,71 @@
 #!/usr/bin/env python3
 
 # The arrow library is used to handle datetimes
-from datetime import datetime
+from datetime import datetime, timedelta
 from logging import Logger, getLogger
 from typing import Optional
 
 import arrow
-from requests import Session
+import pandas as pd
+import pytz
+from requests import Response, Session
+
+from electricitymap.contrib.config import ZONES_CONFIG, ZoneKey
+from electricitymap.contrib.lib.models.event_lists import TotalConsumptionList
+from electricitymap.contrib.parsers.lib.config import refetch_frequency
+from electricitymap.contrib.parsers.lib.exceptions import ParserException
 
 # More info:
 # https://www.bchydro.com/energy-in-bc/our_system/transmission/transmission-system/actual-flow-data.html
 
-timezone = "Canada/Pacific"
+HISTORICAL_LOAD_REPORTS = "https://www.bchydro.com/content/dam/BCHydro/customer-portal/documents/corporate/suppliers/transmission-system/balancing_authority_load_data/Historical%20Transmission%20Data/BalancingAuthorityLoad{0}.xls"
+TIMEZONE = pytz.timezone(ZONES_CONFIG.get(ZoneKey("CA-BC")).get("timezone"))
+PUBLICATION_DELAY = timedelta(days=31)
+
+
+@refetch_frequency(timedelta(days=1))
+def fetch_consumption(
+    zone_key: ZoneKey,
+    session: Optional[Session] = None,
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+):
+    r = session or Session()
+    consumption_list = TotalConsumptionList(logger)
+    if target_datetime is None:
+        start_date = datetime.now(tz=TIMEZONE) - PUBLICATION_DELAY
```

Comment:
Just to be sure we get overlapping data.
```suggestion
        start_date = datetime.now(tz=TIMEZONE) - PUBLICATION_DELAY - timedelta(days=1)
```